suite_report_git/suite_data.py

```diff
@@ -55,8 +55,113 @@ def __init__(self) -> None:
+                    else:
+                        config = ""
+                failed_configs.add(config)
+        return failed_configs
+
+    def read_um_section(self, change: str) -> str:
+        """
+        Read through a UM file searching for line
+        """
+        change = self.temp_directory / "um" / change
+        lines = change.read_text()
+        lines = lines.lower()
+        try:
+            section = re.search(r"this file belongs in section:\s*(\w+)", lines).group(
+                1
+            )
+        except AttributeError:
+            section = "Unknown"
+        return section
+
+    def get_changed_um_section(self) -> Set[str]:
+        """
+        Read through bdiff of UM source and find code owner section for each changed
+        file
+        """
+
+        changed_sections = set()
+        for change_path in self.dependencies["um"]["gitbdiff"]:
+            change = change_path.lower()
+            # First check files which will not have a Code Section Comment
+            if "dependencies.yaml" in change:
+                continue
+            if "configowners.txt" in change or "codeowners.txt" in change:
+                changed_sections.add(change)
+            elif change.startswith("admin"):
+                changed_sections.add("admin")
+            elif change.startswith("bin"):
+                changed_sections.add("bin")
+            elif change.startswith("fcm-make"):
+                changed_sections.add("fcm-make_um")
+            elif change.startswith("fab"):
+                changed_sections.add("fab")
+            elif change.startswith("rose-stem"):
+                changed_sections.add("rose_stem")
+            elif change.startswith("rose-meta"):
+                if "etc/stash" in change:
+                    changed_sections.add("stash")
+                elif "rose-meta.conf" in change:
+                    changed_sections.add("rose-meta.conf")
+                else:
+                    changed_sections.add("upgrade_macros")
+            else:
+                # Read the section from the code comment
+                changed_sections.add(self.read_um_section(change_path))
+
+        return changed_sections
+
+    def get_um_owners(self, filename: str) -> Dict:
+        """
+        Read UM Code Owners file and write to a dictionary
+        """
+
+        fpath = self.temp_directory / "um" / filename
+        owners_text = fpath.read_text()
+
+        in_owners = False
+        owners = {}
+        for line in owners_text.split("\n"):
+            line = line.lower().strip()
+            if (
+                not line
+                or line.startswith("#")
+                or line.startswith("area")
+                or line.startswith("configuration")
+            ):
+                continue
+            if line.startswith("{{{"):
+                in_owners = True
+            elif line.startswith("}}}"):
+                in_owners = False
+            elif in_owners:
```

Comment:
```suggestion
            if in_owners:
```
This needs to be a if statement as the control flow as the if elif block gets exited once the in_owners variable has been set to True and therefore this condition won't be processed when set to True.

Reply:
Thanks Pierre. I think I'm going to disagree on this one, because the format of the files we're looking at. See, https://code.metoffice.gov.uk/trac/um/browser/main/trunk/CodeOwners.txt
The lines with `{{{` or `}}}` in never have a code section as well, so we don't want to look for these in the same line as the delimiters. Admittedly we could probably improve the robustness here, but I'm not sure the extra complexity in code, is worthwhile in this case.

Reply:
I looked to much down the toilet roll and forgot to check the context of this being in a for loop.... [picard.gif]